src/Authress.SDK/Client/HttpClientProvider.cs

```diff
@@ -107,4 +108,30 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             return await base.SendAsync(request, cancellationToken);
         }
     }
+
+    internal class RewriteBaseUrlHandler : DelegatingHandler
+    {
+        private readonly Uri baseUrl;
+
+        public RewriteBaseUrlHandler(HttpMessageHandler innerHandler, string baseUrl) : base(innerHandler)
+        {
+            this.baseUrl = new Uri(baseUrl.EndsWith("/") ? baseUrl : baseUrl + "/");
```

Comment:
I just found `TrimEnd` seems useful here. (`TrimStart` could be used an line 135 as well)

```suggestion
            this.baseUrl = new Uri(baseUrl.TrimEnd('/'));
```